### PR TITLE
Implement str.isprintable() based on CPython

### PR DESCRIPTION
**Pull request by densmirn**

### CODE REVIEW

**Comment by densmirn** [comment]
Could you please add the PR to review queue?

**Comment by stuartarchibald** [comment]
Thanks for the PR. Marked for review.

File: numba/unicode.py [hunk 1]
```diff
@@ -1388,6 +1389,22 @@ def impl(s):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12188-L12213    # noqa: E501
+@overload_method(types.UnicodeType, 'isprintable')
+def unicode_isprintable(data):
+    """Implements UnicodeType.isprintable()"""
+
+    def impl(data):
+        for i in range(len(data)):
```
Comment (stuartarchibald) [inline numba/unicode.py]: The CPython impl has a shortcut for single characters, perhaps implement that?

Comment (densmirn) [reply numba/unicode.py]: As I see the shortcut exactly repeats body of the main loop. Do you think we still need to add the following code to the implementation?
```
if length == 1:
    code_point = _get_code_point(data, 0)
    if _PyUnicode_IsPrintable(code_point):
        return True
    return False
```

Comment (densmirn) [reply numba/unicode.py]: @stuartarchibald What do you think about above comment?

Comment (stuartarchibald) [reply numba/unicode.py]: See https://github.com/numba/numba/pull/4843#discussion_r346745393

**Comment by densmirn** [comment]
Could you please queue the PR for review?

**Comment by stuartarchibald** [comment]
Please resolve code conflicts via merge to permit merge to master, thanks.

**Comment by stuartarchibald** [comment]
Please could you resolve conflicts against master via merge. Thanks.

**Comment by stuartarchibald** [comment]
Merge looks good, thanks.

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch. Thanks!

**Comment by densmirn** [comment]
Resolved all the merge conflicts.

**Pull request merged**